[12.0][IMP] agreement_serviceprofile: add stage

brian10048 opened the pull request:
- Add stage_type to agreement.stage
- Track stage of service profile
- Add Kanban view for service profiles to the dashboard
- Add product_id field to service profile

@max3903 for #349

brian10048 committed:
[IMP] agreement_serviceprofile: add stage

- Add stage_type to agreement.stage
- Track stage of service profile
- Add Kanban view for service profiles to the dashboard
- Add product_id field to service profile

brian10048 commented:
Please excuse the mess! 

@max3903, should I re-use the agreement.stage and add a selection type field for service profile stages like was done with field service stages?

max3903 commented:
Yes please.

El mié., 17 de julio de 2019 22:15, brian10048 <notifications@github.com>
escribió:

> Please excuse the mess!
>
> @max3903 <https://github.com/max3903>, should I re-use the
> agreement.stage and add a selection type field for service profile stages
> like was done with field service stages?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/contract/pull/355?email_source=notifications&email_token=AAHFPMC4SN3G4MAGXZF32DDP77N4TA5CNFSM4IEWLKI2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2HGJFI#issuecomment-512648341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHFPMHY3HNYEGGWJB4CRNLP77N4TANCNFSM4IEWLKIQ>
> .
>

max3903 commented:
@brian10048 I added a product_id field on the design with a domain. Can you implement it please?

max3903 commented:
Ping @kvedder-amplex @barnarj

brian10048 commented:
@max3903 I reworked the stage to use the agreement.stage model

> 
> @brian10048 I added a product_id field on the design with a domain. Can you implement it please?

I'm trying to figure out what you mean on the design.  Could you explain a little more maybe?

max3903 commented:
@brian10048 The service profile is the same to the service product as the serial number to the stockable product.

![Screenshot from 2019-07-12 13-51-07](https://user-images.githubusercontent.com/939952/61465043-f1b7d780-a93c-11e9-8a50-9c04504bbc48.png)

So with a product_id field on the service profile, we can access all the service information defined on the product template.

brian10048 commented:
Thank you @max3903. That helps

Also, not sure what I am missing.  The dashboard kanban view for service profiles is not showing the stages unless there is a record for that stage

![image](https://user-images.githubusercontent.com/25710115/61465569-4875df00-a946-11e9-8688-b23c62e08385.png)

max3903 commented on agreement_serviceprofile/models/agreement_serviceprofile.py:
```diff
@@ -2,14 +2,19 @@
+
     name = fields.Char(string="Name", required=True)
+    stage_id = fields.Many2one('agreement.stage', string="Stage",
+                               default=_default_stage_id, copy=False)
```
```suggestion
                               default=_default_stage_id, copy=False,
                               group_expand='_read_group_stage_ids',)
```

max3903 replied:
@brian10048 That should open all the stages, even without records.

brian10048 replied:
Thank you.  I added the method, but forgot that part in the field definition!

brian10048 commented:
@max3903 I believe this is done now.  Cleaned up into one commit.  Let me know if there is anything else I need to do.

max3903 commented:
:+1: LGTM. Tested on Runbot.

bodedra reviewed (approved):
LTGM :+1: Good work @brian10048 

Not blocking comment: IMO for better maintain and commit history, shouldn't we have two PRs for these ? one for `agreement_legal` and another for `agreement_serviceprofile`

max3903 commented:
/ocabot merge
